src/nestjs-winston-logger.module.ts

```diff
@@ -7,18 +7,44 @@ import {
   getLoggerToken,
 } from "./nestjs-winston-logger.decorator";
 
+export type ContextOverrides = {
+  /**
+   * context is the token used in @InjectLogger.
+   * For example, @InjectLogger(AUTH_LOG) where const AUTH_LOG = "AUTH_LOG"
+   */
+  [context:string]:LoggerOptions
+};
+
+export type LoggerConfig = {
+  /**
+   * Logger configuration used if no overrides provided
+   */
+  defaultConfig:LoggerOptions,
+
+  /**
+   * Provides logger configuration specific to the provided logger context
+   */
+  overrides?: ContextOverrides
+};
+
+
 @Module({})
 export class NestjsWinstonLoggerModule {
-  static forRoot(options: LoggerOptions): DynamicModule {
+  static forRoot(options: LoggerOptions | LoggerConfig): DynamicModule {
```

Comment:
```static forRoot(options: LoggerOptions, overrides?: ContextOverrides): DynamicModule {```
How do you feel about the second parameter?

Reply:
I was wondering if it would be better to do it like the mongoose appraoch.

Reference: https://github.com/nestjs/mongoose/blob/74f7d48fa1ad632270e66de5f4198619b784498a/lib/mongoose-core.module.ts#L33

Ideas:
User will be providing name and specific config to construct a logger, the name will help to identify the responsibility of the logger(e.g. auth logger, req logger) and the config is the LoggerOptions of winston.

These loggers module will be imported into user's module and logger service will be injected into user's provider/service like:
`@Injectable()
export class DemoService {
  constructor(@InjectLogger("auth_logger") private authLogger: NestjsWinstonLoggerService, @InjectLogger("req_logger") private authLogger: NestjsWinstonLoggerService, ) {}
}`

Any thought or concerns about this approach?

Reply:
I'm no comment on this approach, it is good at single responsibility of the logger.

If we adopt this approach, we need to revise `nestjs-winston-logger.decorator.ts` which distinguishes by context name in the current version.

Also, we can optionally pass some metadata into decorator such as context name
```@InjectLogger("auth_logger", { // something } ) private authLogger: NestjsWinstonLoggerService```
something like that, what do you think?

Reply:
@Marcotsept - Re: ```forRoot(options: LoggerOptions, overrides: ContextOverrides = {} )```
I like your approach much better than mine.  Easier to use.  But we just need to make the second parameter optional to ensure backwards compatibility.

@owen1002 - Re: Mongoose approach
I'm afraid I don't understand what you are getting at.  I can't see how the mongoose approach is different from the one I used in terms of injecting into the constructor.   The way you inject the logger with this commit is ```constructor(@InjectLogger("auth_logger") private authLogger: NestjsWinstonLoggerService, ...``` as well.  The only difference is in my example in the read only, I used AUTH_TOKEN which is just an example variable placeholder that consumers can define in order to ensure they don't have typos.  However, it is essentially the same as using a string

@Marcotsept - Re: ```@InjectLogger("auth_logger", { // something })
I like it.  But am having an issue.  Suppose we want to keep @InjectLogger(MyService.name) as an option.  It writes the metadata {..., "service":"MyService", ...} to the log.  If we do @InjectLogger("auth_logger", { service:MyService.name }) then we can do something similar by just making sure the second parameter is present so that we know the first is not the service name.  But then if I want to do just @InjectLogger("auth_logger"), I'm unable to distinguish that it is not a class name and will end up writing {..., "service":"auth_logger", ...}.  Any suggestions on this?

Reply:
One other question.  I don't think "overrides" is the right term for the way I have it implemented.  It is a bit unclear as to whether or not it just overrides common settings from the defaultOptions or completely replaces the defaultOptions.  Any suggestions on a good alternative or is it best to handle with just better docs.